package/kedro_viz/api/rest/responses.py

```diff
@@ -273,19 +326,151 @@ def get_default_response() -> GraphAPIResponse:
+    """API response for `/api/nodes/node_id`."""
+    node = data_access_manager.nodes.get_node_by_id(node_id)
+    if not node:
+        return JSONResponse(status_code=404, content={"message": "Invalid node ID"})
 
-        Returns:
-            A bytes object containing the JSON to write.
+    if not node.has_metadata():
+        return JSONResponse(content={})
 
-        """
-        return orjson.dumps(
-            content,
-            option=orjson.OPT_INDENT_2
-            | orjson.OPT_NON_STR_KEYS
-            | orjson.OPT_SERIALIZE_NUMPY,
+    if isinstance(node, TaskNode):
+        return TaskNodeMetadata(node)
+
+    if isinstance(node, DataNode):
+        return DataNodeMetadata(node)
+
+    if isinstance(node, TranscodedDataNode):
+        return TranscodedDataNodeMetadata(node)
+
+    return ParametersNodeMetadata(node)
+
+
+def get_selected_pipeline_response(registered_pipeline_id: str):
+    """API response for `/api/pipeline/pipeline_id`."""
+    if not data_access_manager.registered_pipelines.has_pipeline(
+        registered_pipeline_id
+    ):
+        return JSONResponse(status_code=404, content={"message": "Invalid pipeline ID"})
+
+    modular_pipelines_tree = (
+        data_access_manager.create_modular_pipelines_tree_for_registered_pipeline(
+            registered_pipeline_id
+        )
+    )
+
+    return GraphAPIResponse(
+        nodes=data_access_manager.get_nodes_for_registered_pipeline(  # type: ignore
+            registered_pipeline_id
+        ),
+        edges=data_access_manager.get_edges_for_registered_pipeline(  # type: ignore
+            registered_pipeline_id
+        ),
+        tags=data_access_manager.tags.as_list(),
+        layers=data_access_manager.get_sorted_layers_for_registered_pipeline(
+            registered_pipeline_id
+        ),
+        pipelines=data_access_manager.registered_pipelines.as_list(),
+        selected_pipeline=registered_pipeline_id,
+        modular_pipelines=modular_pipelines_tree,  # type: ignore
+    )
+
+
+def get_package_compatibilities_response(
+    package_name: str = _FSSPEC_PACKAGE_NAME,
+    compatible_version: str = _FSSPEC_COMPATIBLE_VERSION,
+):
+    """API response for `/api/package_compatibility`."""
+    package_version = get_package_version(package_name)
+    is_compatible = packaging.version.parse(package_version) >= packaging.version.parse(
+        compatible_version
+    )
+    return PackageCompatibilityAPIResponse(
+        package_name=package_name,
+        package_version=package_version,
+        is_compatible=is_compatible,
+    )
+
+
+def write_api_response_to_fs(file_path: str, response: Any, remote_fs: Any):
+    """Encodes, enhances responses and writes it to a file"""
+    jsonable_response = jsonable_encoder(response)
+    encoded_response = EnhancedORJSONResponse.encode_to_human_readable(
+        jsonable_response
+    )
+
+    with remote_fs.open(file_path, "wb") as file:
+        file.write(encoded_response)
+
```

Comment:
I was a bit surprised it is using "wb" instead of "w". I expect the result of `encode_to_human_readable` should be text that is readable, but seems like it is returning `bytes`?

Reply:
yes, it's currently returning bytes. I will have to create an issue to figure the best way to make this more clearer. But it would be out of scope for the current one.